### ISSUE TITLE
Different version of Entity Framework Core and project target framework

### ISSUE THREAD

**Issue by ProH4Ck** [ErikEJ/EFCorePowerTools#2132]
Hi @ErikEJ ,
I upgraded a project from net6 to net8 but now EF Core Power Tools forces me to use EF Core 8.
The project references EF Core 7 because I need to connect to a legacy Oracle database 11 which is not supported by latest Oracle EF Core provider for EF Core 8.
I reproduced the issue on a sample console:

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/8057670/af938128-c835-4f81-9ae0-95c42ff31999)

As you can see EF Core 7 is installed but EF Core Power Tool it's locked to EF Core 8.
It works only if I downgrade project to net7... in this case I can select between EF Core 6 and EF Core 7.

Is there a technical reason why net8 is locked on EF Core 8? Da you think is possible read the `Microsoft.EntityFrameworkCore` version from csproj and use this version in the tool?

Thanks,

Emanuele

**Comment by ErikEJ** [comment]
.NET 8 only supports/works with EF Core 8

**Comment by ProH4Ck** [comment]
> .NET 8 only supports/works with EF Core 8

I think it's the opposite: EF Core 8 only works on net8:

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/8057670/23801a1a-80f7-40e1-9251-a41285fe563d)

EF Core 7.0.15 actually supports net8:

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/8057670/8a888c2b-31d3-4661-8d62-f00f03ab23b6)

I completed the console and project targeting net8 works perfectly with EF Core 7:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.15" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="7.0.15" />
  </ItemGroup>

</Project>
```

```csharp
using EfptConsole;

var dbContext = new SqlDbContext();

var users= dbContext.Users.ToList();

Console.WriteLine("There are " + users.Count + " users on DB");
```

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/8057670/096f7513-22ce-4258-91e4-25629bbfca86)

**Comment by ErikEJ** [comment]
@ajcvickers @roji Is this a supported scenario?

**Comment by ajcvickers** [comment]
@ErikEJ There's this: https://github.com/dotnet/efcore/issues/32782 But I think after this fix the .NET 8 tool will still need the .NET 8 SDK.

**Comment by ErikEJ** [comment]
@ajcvikcers But otherwise it is OK to use EF Core 7 (and maybe even 6) with .NET 8?

**Comment by ajcvickers** [comment]
Yep. I only have the .NET 8 SDK installed, but I have been running repros from 6, 7, and 8.

**Comment by ErikEJ** [comment]
@ajcvickers Thanks, and all these repros target net8.0 ?

**Comment by ajcvickers** [comment]
Yes. But I would still use .NET 8 to build even if I were targeting .NET 6 or .NET 7. Of course, I would then need the appropriate runtime installed to run the app. Rather than doing that, I usually just switch the project to target .NET 8. Haven't seen any issues yet.

**Comment by ErikEJ** [comment]
@ajcvickers Thanks, I will allow projects targeting net8.0 to use EF Core 7 as well then!

**Comment by ErikEJ** [comment]
@ProH4Ck 

I implemented a fix for this in the [latest daily build](https://www.vsixgallery.com/extension/f4c4712c-ceae-4803-8e52-0e2049d5de9f), (build 98 or later) would be grateful if you could try it out.

**Comment by ProH4Ck** [comment]
Ok I'll try immediatly on the real project with the daily build.

**Comment by ProH4Ck** [comment]
I cannot see build 98: 
![image](https://github.com/ErikEJ/EFCorePowerTools/assets/8057670/3a69bf8f-a923-4d5b-85d7-b6529a5ab151)

**Comment by ProH4Ck** [comment]
It's working!

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/8057670/e8fd6247-09a8-4407-8ee3-751156efa8f8)

**Comment by ErikEJ** [comment]
@ProH4Ck Wonderful!
